harness/determined/_core/_distributed.py

```diff
@@ -215,11 +219,13 @@ def _zmq_gather(self, stuff: Any) -> Optional[List]:
             return [stuff]
         logging.debug(f"Worker {self.get_rank()} beginning zmq gather.")
         if self._is_chief:
-            worker_stuff, _ = self._chief_zmq.gather_with_polling(lambda: None)
+            worker_stuff_ranked, _ = self._chief_zmq.gather_with_polling(lambda: None)
+            worker_stuff_ranked.sort(key=_tuple_key_function)
+            worker_stuff = [value for value, _ in worker_stuff_ranked]
```

Comment:
If you reorder the worker tuple, this could be:
```python
worker_stuff_ranked, _ = self._chief_zmq.gather_with_polling(lambda: None)
worker_stuff = [x[1] for x in sorted(worker_stuff_ranked)]
```

The way you have it ok though, so please consider this non-blocking.

Reply:
I don't want to pass on an opportunity to simplify the code.